nmtwizard/preprocess/prepoperator.py

```diff
@@ -22,105 +22,147 @@ def get_operator_params(config):
     return config
 
 
+class ProcessType(object):
+      """Type of processing pipeline.
+
+      Possible values are:
+        * ``TRAINING``
+        * ``INFERENCE``
+        * ``POSTPROCESS``
+      """
+
+      TRAINING = 0
+      INFERENCE = 1
+      POSTPROCESS = 2
+
+
 class Pipeline(object):
     """Pipeline for building and applying pre/postprocess operators in order."""
 
-    def __init__(self, config, process_type, preprocess_exit_step):
-        # Current state of preprocessing pipeline.
-        # Passed to and modified by operator initializers if necessary.
-        self.start_state = { "src_tokenizer" : None,
-                             "tgt_tokenizer" : None }
+    def __init__(self, config, preprocess_exit_step):
+        self._build_state = None
+        self._process_type = ProcessType.TRAINING
+        self._build_preprocess_pipeline(config, preprocess_exit_step)
 
-        self._build_state = dict(self.start_state)
+
+    def _build_preprocess_pipeline(self, config, preprocess_exit_step=None):
 
         self._ops = []
 
         preprocess_config = config.get("preprocess")
         if preprocess_config:
             for i, op in enumerate(preprocess_config):
-                operator = self._build_operator(op, process_type)
-                if operator and operator.is_applied():
+                operator = self._build_operator(op)
+                if operator:
```

Comment:
At this point we could know if the operator will be applied or not (cf. `is_applied_for` method). Do we still want to add it in the list of ops?